modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -3428,7 +3428,12 @@ def func(df) -> np.ndarray:
             return ser.cat.codes.to_frame(name=MODIN_UNNAMED_SERIES_LABEL)
 
         res = self._modin_frame.apply_full_axis(
-            axis=0, func=func, new_columns=[MODIN_UNNAMED_SERIES_LABEL]
+            axis=0,
+            func=func,
+            new_index=self._modin_frame._index_cache,
+            new_columns=[MODIN_UNNAMED_SERIES_LABEL],
+            new_row_lengths=self._modin_frame._row_lengths_cache,
+            new_column_widths=self._modin_frame._column_widths_cache,
```

Comment:
I'm still -1 about introducing these parameters as already explained in [this comment](https://github.com/modin-project/modin/pull/5587#discussion_r1094484673).

I still believe that we can precompute new row/column lengths cache based only on the `new_index` and `new_columns` arguments.

Reply:
We can't correctly precompute new row/column lengths cache based only on the `new_index` and `new_columns` arguments for all cases. E.g., when performing an operation along `axis=1`, we can precompute new column widths using the "standard" partitioning but we can't correctly precompute new row lengths for all cases because an operation may change the shape of a partition along axis=0 even if we perform the operation along `axis=1`. So even now we have a similar problem when `keep_partitioning=False` and we precompute new row/column lengths which may be incorrect because of what I said in the previous sentence. However, that hasn't bited us yet.

So I can replace the arguments to `preserve_row_lengths` and `preserve_column_widths` if this makes sense to you.

Reply:
I just remembered that we have a dedicated operator for such cases. Can we use [`.fold()`](https://github.com/modin-project/modin/blob/d724802391524ad21847722d3d92f8448233f500/modin/core/dataframe/pandas/dataframe/dataframe.py#L1796) here? I think it has to be used in exactly the cases we're experiencing here, it applies the function along the whole axis an assumes that the data shape was not changed thus propagating all of the metadata to the resulting frame.

Reply:
the only problem with `.fold` is that it doesn't allow column renaming...
but maybe it's worth adding it?

Reply:
Good catch! I think we can use `fold` operator here, changed in the latest commit. However, it doesn't resolve the issue of `apply_full_axis`, which may bite us in future.